### PR TITLE
YAML inventory doc: add info about 'all' group, remove unused option

### PR DESCRIPTION
**Pull request by pilou-**
##### SUMMARY

- Add information about `all` group
- ~~remove unused option `yaml_extensions`~~

##### ISSUE TYPE
 - Docs Pull Request

##### COMPONENT NAME
YAML inventory plugin

##### ANSIBLE VERSION
```
ansible-playbook 2.4.0 (devel 483df9c5f8) last updated 2017/06/06 23:50:25 (GMT +200)
```

### CODE REVIEW

**Comment by gundalow** [comment]
@jhoekx I believe you've made functional changes to this before. It would be good to get your review input

**Comment by acozine** [comment]
@pilou- thanks for reviewing the documentation. I'm not sure what your change to line 24 means. Could you describe the experience that led you to suggest this change? Did you get failures or strange behavior because your inventory file had no group called `all`? Or did your playbook run on different machines than you expected?

**Comment by pilou-** [comment]
@acozine it comes from a discussion on #ansible-devel (more than one year ago, it was the beginning of inventory plugins):

> 06-06-2017 22:56:57 > Pilou: bcoca: with YAML inventory plugin, is the 'all' group mandatory ?
06-06-2017 22:57:52 < bcoca: no, but i recommend it
06-06-2017 22:58:09 < bcoca: will represent actual hierchy that inventory becomes
06-06-2017 22:58:27 < bcoca: also, only way to set 'all:vars'

@bcoca is that still true? If so, `'all:vars' should be reformulated.

This PR might has been obsoleted by #26770.

**Comment by bcoca** [comment]
not really `all:vars`, that was constraint of irc not being multiline

```yaml
all:
      vars:
```
what i meant, it is only way to set vars for the 'all' group inside the inventory file

File: lib/ansible/plugins/inventory/yaml.py [hunk 1]
```diff
@@ -9,16 +9,15 @@
...
         - Vars entries are normal group vars.
         - "Children are 'child groups', which can also have their own vars/hosts/children and so on."
         - File MUST have a valid extension, defined in configuration
     notes:
-        - It takes the place of the previously hardcoded YAML inventory.
-        - To function it requires being whitelisted in configuration.
-        - Beginning with the C(all) group is the only way to set vars for the
-          C(all) group inside the inventory file.
+        - Replaces the previously hardcoded YAML inventory.
+        - Must be whitelisted in configuration to function.
```
Comment (bcoca) [inline lib/ansible/plugins/inventory/yaml.py]: we can remove these 2 lines, by now no one remembers it used to be hardcoded and it is whitelisted by default (also docs explain this in main page).

Comment (acozine) [reply lib/ansible/plugins/inventory/yaml.py]: cool - I removed the line about how it used to be hardcoded and changed the note about whitelisting

**Pull request merged**